src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -78,33 +79,16 @@ internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
             return dt.ToLocalTime();
         }
 
+        private static long s_bootTimeTicks;
         /// <summary>Gets the system boot time.</summary>
         private static DateTime BootTime
         {
             get
             {
-                // '/proc/stat -> btime' gets the boot time.
-                // btime is the time of system boot in seconds since the Unix epoch.
-                // It includes suspended time and is updated based on the system time (settimeofday).
-                const string StatFile = Interop.procfs.ProcStatFilePath;
-                string text = File.ReadAllText(StatFile);
-                int btimeLineStart = text.IndexOf("\nbtime ", StringComparison.Ordinal);
-                if (btimeLineStart >= 0)
-                {
-                    int btimeStart = btimeLineStart + "\nbtime ".Length;
-                    int btimeEnd = text.IndexOf('\n', btimeStart);
-                    if (btimeEnd > btimeStart)
-                    {
-                        if (long.TryParse(text.AsSpan(btimeStart, btimeEnd - btimeStart), out long bootTimeSeconds))
-                        {
-                            return DateTime.UnixEpoch + TimeSpan.FromSeconds(bootTimeSeconds);
-                        }
-                    }
-                }
-
-                return DateTime.UtcNow;
-            }
-        }
+               Interlocked.CompareExchange(ref s_bootTimeTicks, Interop.Sys.GetBootTimeTicks(), 0);
+               return new DateTime(s_bootTimeTicks);
```

Comment:
this implementation is thread safe, but not optimal from perf perspective as we always perform two sys-calls even if the vale has been already cached

```suggestion
               long bootTimeTicks = Interlocked.Read(ref s_bootTimeTicks);
               if (bootTimeTicks == 0)
               {
                    bootTimeTicks = Interop.Sys.GetBootTimeTicks();
                    long oldValue = Interlocked.CompareExchange(ref s_bootTimeTicks, bootTimeTicks, 0);
                    if (oldValue != 0) // a different thread has managed to update the ticks first
                    {
                        bootTimeTicks = oldValue; // consistency
                    }
               }
               return new DateTime(bootTimeTicks);
```

Reply:
Looks complicated for something not meant for hot path. Going by the previous implementation, perf is not the concern for this property, we only want a stable result across accesses.

Reply:
I haven't measured it but is trading `Interlocked.CompareExchange(ref s_bootTimeTicks, bootTimeTicks, 0);` every call with `Interlocked.CompareExchange(ref s_bootTimeTicks, 0, 0);` every call a win? This is what Read(ref long) does:
https://github.com/dotnet/runtime/blob/9158543338acd47019ffb6ec634a273a9e28cb99/src/coreclr/System.Private.CoreLib/src/System/Threading/Interlocked.CoreCLR.cs#L205-L206

Reply:
@am11 the most expensive thing is the call to `Interop.Sys.GetBootTimeTicks()`, which in my proposal is done only if the field has not been initialized yet

Reply:
> This is what Read(ref long) does

It doesn't need an Interlocked.Read, just a Volatile.Read, which on x64 is basically free and on arm is a cheaper barrier.

Reply:
Ah, I missed the upfront inteop call cost.
Thanks for the Volatile.Read tidbit. :)